test/test-tcp-rst.c

```diff
@@ -47,9 +47,9 @@ static void read_cb(uv_stream_t* t, ssize_t nread, const uv_buf_t* buf) {
   ASSERT_PTR_EQ((uv_tcp_t*) t, &tcp);
   ASSERT_EQ(nread, UV_ECONNRESET);
 
-  int fd;
+  uv_os_fd_t fd;
   ASSERT_OK(uv_fileno((uv_handle_t*) t, &fd));
-  uv_handle_type type = uv_guess_handle(fd);
+  uv_handle_type type = uv_guess_handle((uv_file) fd);
```

Comment:
Shouldn't you need to allocate the converted object (with uv_open_osfhandle), rather than directly casting this? It seems strange this has ever worked.

> Man, this uv_os_fd_t vs uv_file.

The difference is gone in v2, as soon as you can convince nodejs to upgrade...

Reply:
> Shouldn't you need to allocate the converted object (with uv_open_osfhandle), rather than directly casting this? It seems strange this has ever worked.

I can double-check!